packages/connect-webextension/src/channels/serviceworker-window.ts

```diff
@@ -65,9 +75,9 @@ export class ServiceWorkerWindowChannel<
                 }
 
                 // eslint-disable-next-line no-restricted-globals
-                if (origin === self.origin) {
-                    return;
-                }
+                // if (origin === self.origin) {
+                //     return;
+                // }
```

Comment:
I am not sure about the reason for this. Is it to avoid messages from the same channel?

In the case of webextension frontend communicating with service worker of the same webextension the origin is the same.

Reply:
I'd say yes. not sure yet if it can be safely removed, maybe yes, maybe it had some reason.

we should evaluate if this is the way to go or possibly creating a new channel serviceworker-extensionwindow woud;t be better. or possibly another option to constructor which would tell this class whether communication happens on the same origin? 

just ideas

Reply:
So I added this new flag `allowSelfOrigin`  to allow both situations https://github.com/trezor/trezor-suite/pull/10751/files#diff-d5893bf06be6660fde64670a6caf03bfb1b737d79b982e199bf7a6800b1b318cR25